packages/amplication-server/src/core/git/git.provider.service.ts

```diff
@@ -358,71 +371,75 @@ export class GitProviderService {
     args: CompleteGitOAuth2FlowArgs
   ): Promise<GitOrganization> {
     const { code, gitProvider, workspaceId } = args.data;
+    // provider properties to instantiate the git client service
+    const initialProviderProperties = {
+      installationId: null,
+      clientId: this.clientId,
+      clientSecret: this.clientSecret,
+    };
     const gitClientService = await new GitClientService().create(
       {
         provider: gitProvider,
-        installationId: null,
-        clientId: this.clientId,
-        clientSecret: this.clientSecret,
+        providerProperties: initialProviderProperties,
       },
       this.logger
     );
     try {
-      const {
-        accessToken,
-        refreshToken,
-        expiresIn,
-        tokenType,
-        scopes,
-        userData: { name, uuid },
-      } = await gitClientService.completeOAuth2Flow(code);
+      const { providerProperties, useGroupingForRepositories } =
+        await gitClientService.completeOAuth2Flow(code);
 
       return this.prisma.gitOrganization.upsert({
         where: {
           // eslint-disable-next-line @typescript-eslint/naming-convention
           provider_installationId: {
             provider: gitProvider,
-            installationId: uuid,
+            installationId: providerProperties.uuid,
           },
         },
         create: {
           provider: gitProvider,
-          installationId: uuid,
-          name,
+          installationId: providerProperties.uuid,
+          name: providerProperties.username,
           type: EnumGitOrganizationType.User,
+          useGroupingForRepositories,
           workspace: {
             connect: {
               id: workspaceId,
             },
           },
-          providerProperties: {
-            username: name,
-            uuid,
-            accessToken,
-            refreshToken,
-            expiresIn,
-            tokenType,
-            scopes,
-          },
+          providerProperties,
         },
         update: {
-          name: name,
-          providerProperties: {
-            username: name,
-            uuid,
-            accessToken,
-            refreshToken,
-            expiresIn,
-            tokenType,
-            scopes,
-          },
+          name: providerProperties.username,
+          providerProperties,
         },
       });
     } catch (error) {
       this.logger.error(error);
     }
   }
 
+  async getGitGroups(args: GitGroupArgs): Promise<PaginatedGitGroup> {
+    const organization = await this.getGitOrganization({
+      where: {
+        id: args.where.organizationId,
+      },
+    });
+    const gitClientService = await new GitClientService().create(
+      {
+        provider: EnumGitProvider[organization.provider],
+        providerProperties: {
+          ...JSON.parse(JSON.stringify(organization.providerProperties)),
+          clientId: this.clientId,
+          clientSecret: this.clientSecret,
```

Comment:
I don't see why we need to send a special object instead of just passing the providerProperties as it is. 
What are clientId and clientSecret? Where do we get it from?

Reply:
These values come from environment variables. We talked about it. The git-utils is a library, and it shouldn't rely on env variables, so I moved the `process.env` part from the lib to amplication-server

Reply:
so we need to think on how to pass them. It should not be passed as part of the providerProperties since the client of the git-util should not  know about the structure and type of the param